### PR TITLE
mat: test that SymEigen eigenvalues are in ascending order

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

### CODE REVIEW

File: mat/eigen_test.go [hunk 1]
```diff
@@ -160,6 +160,13 @@ func TestSymEigen(t *testing.T) {
 					t.Errorf("Eigenvalue does not match")
 				}
 			}
+
+			// Check that the eigenvalues are in ascending order.
```
Comment (kortschak) [inline mat/eigen_test.go]: `if !sort.Float64sAreSorted(es.values) {`

Comment (vladimir-ch) [reply mat/eigen_test.go]: Oh. Done.

**Comment by kortschak** [comment]
Is the same thing true of Eigen? I don't think `lapacke.Dgeev` says so.

**Comment by vladimir-ch** [comment]
`Dgeev` gives complex eigenvalues in general but there is no natural ordering for complex numbers. We could do _something_ but I think it's not worth it.

**Comment by kortschak** [comment]
Good point.

**Pull request merged**